### PR TITLE
Implement JSON Feed output

### LINKED ISSUE #165: Add JSON Feed support
JSON has become the developers’ choice for APIs, and that developers will often go out of their way to avoid XML. RSS is nice as it has widespread support, but JSON is easier to use in ad-hoc applications, especially on frontend (point in case: I'm trying to add a small widget with latest articles from my blog right now).

Here's [specification](https://www.jsonfeed.org/version/1.1/) and [mapping from RSS](https://www.jsonfeed.org/mappingrssandatom/). Would you be interested in a PR implementing this?

### PR DESCRIPTION
**Pull request by notpushkin**
This currently [almost passes](https://validator.jsonfeed.org/?url=https%3A%2F%2Fgistcdn.rawgit.net%2Fnotpushkin%2F45cba83fe486d836a5abef8885612b72%2Fraw%2F6356c6a66be41c86cfb2f0b6fc98d8a1f4332773%2Fjsonfeed.json) JSON Feed validator. The few things it complains about are `null` values that should instead be removed from the feed. Apart from that and some code review and more thorough testing I think we're good to go!

Closes #165

### CODE REVIEW

File: mkdocs_rss_plugin/util.py [hunk 1]
```diff
@@ -593,9 +594,51 @@ def filter_pages(pages: list, attribute: str, length: int) -> list:
...
+                    "title": item.get("title"),
+                    "content_html": item.get("description"),
+                    "image": (item.get("image") or (None,))[0],
+                    entry_date_key: item.get("pubDate3339"),
+                    "authors": [{"name": name} for name in (item.get("authors") or ())],
+                    "tags": item.get("categories"),
+                }
+                for item in feed.get("entries", ())
+            ],
+        }
```
Comment (notpushkin) [inline mkdocs_rss_plugin/util.py:644]: I was a bit hesitant as to whether put this in a helper method like this or a template like this:

```twig
{{ {
  "version": "https://jsonfeed.org/version/1",
  "title": feed.title,
} | tojson }}
```

The template approach feels more in line with the current system, but on the other hand spawning Jinja just to make one JSON feels a bit weird. @Guts What do you think?

Comment (Guts) [reply mkdocs_rss_plugin/util.py:644]: I get your dilemma. Let's move on with this implementation using JSON module and when a fully featured customization mechanism will be needed, we'll reconsider that.

**Comment by Guts** [comment]
Thanks for your work, it's really cool!

Last steps before complete review and merge:

- [ ] add unit tests including compliance tests and some special cases (special chars, etc.)
- [ ] complete documentation

**Comment by Guts** [comment]
@notpushkin any chance to see you fit the missing points in the next days https://github.com/Guts/mkdocs-rss-plugin/pull/177#issuecomment-1479260335?

**Comment by notpushkin** [comment]
@Guts sorry for the wait! If you're up to it, please do finish it up for me – my mind is completely somewhere else right now, so I can't say for sure when I'll be able to do it :-( Again extremely sorry for this half-assed PR, but I hope it helps somebody!

**Comment by Guts** [comment]
Hi @notpushkin,

Better late than never, I took the time to achieve your feature.
It still lacks of compliance validation but I think we can move forward and improve it later.

**Comment by notpushkin** [comment]
Thank you! Sorry for the radio silence, I've had a lot of things going on this year, way over what I can usually handle :')

Hope this feature helps other users and somebody will be motivated enough to make it fully compliant! It still should be pretty useful in the meantime.

**Comment by Guts** [comment]
No worries! It was already great to come to this PR. Thanks again.

I'll make a release in a few days.

**Pull request merged**